Platform/CoffeelakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -267,18 +269,21 @@ PlatformIdInitialize (
        (CpuDid == V_SA_DEVICE_ID_CFL_DT_3) ||
        (CpuDid == V_SA_DEVICE_ID_CFL_DT_4))) {
       SetPlatformId (PLATFORM_ID_CFL_S);  // CFL-S
+      SetPlatformBomId (CPU_ID_CFL_S);
     } else if (CpuFamilyModel == CPUID_FULL_FAMILY_MODEL_COFFEELAKE_DT_HALO &&
       ((CpuDid == V_SA_DEVICE_ID_CFL_HALO_1) ||
        (CpuDid == V_SA_DEVICE_ID_CFL_HALO_2) ||
        (CpuDid == V_SA_DEVICE_ID_CFL_HALO_3) ||
        (CpuDid == V_SA_DEVICE_ID_CFL_HALO_IOT_1))) {
       SetPlatformId (PLATFORM_ID_CFL_H);  // CFL-H
```

Comment:
Why do we need duplicated one? We have PlatformId, BomId(cpu id?), PchId(LP or H).
FYI, all platforms are using the same CFL cpu.

Reply:
I think the problem here is that we used (CPU ID + PCH ID) combination as PlatformId.  However, if different boards have same CPU and PCH, then current code will detect them as the same PlatformId.   I believe we  might want a mechanism to address this.